apps/mcpserver/src/auth/oauth.ts

```diff
@@ -0,0 +1,135 @@
+import type { IncomingMessage } from "http";
+
+import { getLogger } from "@logtape/logtape";
+import { JWTIssuer, OAuthProxy } from "fastmcp/auth";
+import * as assert from "node:assert/strict";
+
+import { getConfig } from "../config/auth.js";
+import { region } from "../config/aws.js";
+import { DynamoDBStore } from "./ddbTokenStore.js";
+import { verifyGithubUser } from "./github.js";
+
+const logger = getLogger(["mcpserver", "oauth"]);
+
+let authProxy: OAuthProxy;
+const authConfig = await getConfig();
+
+/**
+ * Gets the initialized OAuth provider.
+ * @throws Error if the provider hasn't been initialized yet.
+ */
+export async function getOAuthProvider(): Promise<OAuthProxy> {
+  if (!authProxy) {
+    authProxy = await initializeOAuthProvider();
+  }
+  return authProxy;
+}
+
+/**
+ * Initializes the GitHub OAuth provider with the client secret.
+ * Must be called before accessing the authProxy.
+ */
+export async function initializeOAuthProvider(): Promise<OAuthProxy> {
+  logger.debug("Fetching GitHub client ID from SSM...");
+  const clientId = await authConfig.GITHUB_CLIENT_ID;
```

Comment:
```suggestion
  const clientId = authConfig.GITHUB_CLIENT_ID;
```
not needed. `authConfig` is not a Promise here